src/components/Avatar/Avatar.tsx

```diff
@@ -175,6 +162,18 @@ const AvatarStatusItems: FC<BaseAvatarProps> = React.forwardRef(
                 ? { 'aria-label': statusItemProps.ariaLabel }
                 : {})}
             >
+              {statusItemProps.text ? (
+                <span
+                  style={{
+                    fontSize: `calc(2px + ${statusItemProps.size})`,
+                    color: statusItemProps.color,
+                  }}
+                >
+                  {statusItemProps.text}
```

Comment:
This may be problematic if the user has `40,000` unread messages, etc. and won't scale. As long as it's compressed to `40k` we should be good. Though your implementation here should ensure a max width to avoid an edge case where a dev allows a large number.

Reply:
<img width="159" alt="image" src="https://user-images.githubusercontent.com/110680523/223967779-61565699-a36f-487c-ac01-2dd00bcdb0ba.png">

Enforcing max-width will result in visual loss of information.
Consider the three cases in the above image:
1. All information is visible. But, the width is more. User can report in case it is a blocker
2. Only partial information is visible. User is very likely to report it. Two solutions possible to overcome this issue:
    1. use something like `200k` - not possible from within the component as we do not know what text the dev will want to have in it
    2. show a tooltip - not suggested as it will complicate the UI
3. All information is visible and the width is also decent enough.

If we enforce max-width then case 2 will always occur and be an issue for the end user.
Occurrence of case 1 and 3 depends on the developer.

So, I think we should not restrict the width.

Reply:
What about a small Avatar? This will not scale.

Reply:
In the case of a large number, it would be best to just not display the number via a typeScript helper method that checks the num length.

Reply:
@ypike-eightfold Please provide some guidance in Figma.